infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -99,17 +91,38 @@ public LogReplicationConfigManager(CorfuRuntime runtime, String pluginConfigFile
         this.corfuStore = new CorfuStore(runtime);
 
         initStreamNameFetcherPlugin();
-        openTables();
+        openVersionTable();
+        unregisterStreamsNameTable(runtime);
     }
 
-    private void openTables() {
+    /**
+     * Helper method for remove Streams_Name_Table and its record in registry table.
+     */
+    private void unregisterStreamsNameTable(CorfuRuntime runtime) {
+        String tableNameString = "LogReplicationStreams";
+        TableName tableNameKey = TableName.newBuilder()
+                        .setTableName(tableNameString)
+                        .setNamespace(CORFU_SYSTEM_NAMESPACE)
+                        .build();
+        if (runtime.getTableRegistry().getRegistryTable().containsKey(tableNameKey)) {
+            try {
+                log.debug("Deleting record of Streams_Name_Table from registry table");
+                // Make sure the tableMap in TableRegistry contains the table
+                runtime.getTableRegistry().openTable(CORFU_SYSTEM_NAMESPACE, tableNameString,
+                        TableInfo.class, Namespace.class, Uuid.class, TableOptions.builder().build());
+                runtime.getTableRegistry().deleteTable(CORFU_SYSTEM_NAMESPACE, tableNameString);
```

Comment:
I think this and the remove() should be done in a transaction.  Also, it is not enough to just delete the entries from these 2 places.  Check PR 3230 which also removes the Protobufdescriptors.

I think we need such a method in TableRegistry which would be invoked from here.